apps/www/registry/default/ui/multi-select.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { cn } from 'lib/utils';
+
+import { Check, X, ChevronsUpDown } from 'lucide-react';
+import { Button } from '@/registry/default/ui/button'
+import {
+    Command,
+    CommandEmpty,
+    CommandGroup,
+    CommandInput,
+    CommandItem,
+} from '@/registry/default/ui/command';
+import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+} from '@/registry/default/ui/popover';
+import { Badge } from '@/registry/default/ui/badge';
+
+export type OptionType = Record<'value' | 'label', string>;
+
+interface MultiSelectProps {
+    options: OptionType[];
+    selected: string[];
+    onChange: React.Dispatch<React.SetStateAction<string[]>>;
+    className?: string;
+    placeholder?: string;
+}
+
+const MultiSelect = React.forwardRef<HTMLButtonElement, MultiSelectProps>(
+    ({ options, selected, onChange, className, ...props }, ref) => {
+        const [open, setOpen] = React.useState(false);
+
+        const handleUnselect = (item: string) => {
+            onChange(selected.filter((i) => i !== item));
+        };
+
+        return (
+            <Popover open={open} onOpenChange={setOpen}>
+                <PopoverTrigger asChild className={className}>
+                    <Button
+                        ref={ref}
+                        variant="outline"
+                        role="combobox"
+                        aria-expanded={open}
+                        className={`w-full justify-between ${selected.length > 1 ? 'h-full' : 'h-10'
+                            }`}
+                        onClick={() => setOpen(!open)}
+                    >
+                        <div className="flex flex-wrap items-center gap-1">
+                            {selected.map((item) => (
+                                <Badge
+                                    variant="outline"
+                                    key={item}
+                                    className="mb-1 mr-1"
+                                    onClick={() => handleUnselect(item)}
+                                >
+                                    {item}
+                                    <span
```

Comment:
Could this be a button?

Reply:
added BUtton with asChild flag, to avoid "<button> cannot appear as a descendant of <button>"